src/components/Correlation/index.js

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { uuid } from "../../utils";
+
+const createCorrelation = () => {
+  return {
+    lifecycle: {
+      onBeforeEvent(event, options) {
+        return Promise.resolve(options.correlationID).then(correlationID => {
```

Comment:
Note that we needed to get `correlationID` from the `options` object in order to handle it and attach it appropriately to the event. Previously, components didn't have access to `options` from within their `onBeforeEvent`, so I changed DataCollector to pass `options` along.

The alternative to this is to make DataCollector in charge of handling the `correlationID` and attaching it appropriately to the event. The role and responsibility of the DataCollector isn't well-defined to me so I wasn't really sure if that's what we want to do. It seemed to me like breaking out correlation ID handling into its own component made sense. Thoughts?